### PR TITLE
A4A non-AMP creative render defaults to safeframe for iOS

### PR DESCRIPTION
**Pull request by keithwrightbos**
Due to caching issue on iOS (does not properly re-user XHR GET cached response when set as source of cross domain iframe), make safeframe rendering the default it rendering not specified in response header and running on iOS.

### CODE REVIEW

**Comment by tdrl** [comment]
LGTM.  Over to @lannka for AMP review.

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 1]
```diff
@@ -343,6 +348,11 @@ export class AmpA4A extends AMP.BaseElement {
           // https://github.com/ampproject/amphtml/issues/5614
           this.experimentalNonAmpCreativeRenderMethod_ =
               fetchResponse.headers.get(RENDERING_TYPE_HEADER);
+          let platform;
+          if (!this.experimentalNonAmpCreativeRenderMethod_ &&
+            (platform = platformFor(this.win)) && platform.isIos()) {
```
Comment (lannka) [inline extensions/amp-a4a/0.1/amp-a4a.js]: `platformFor` never returns `null`, so it's safe to do

```
if (!this.experimentalNonAmpCreativeRenderMethod_ && platformFor(this.win).isIos()) {
```

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 2]
```diff
@@ -171,7 +172,11 @@ export class AmpA4A extends AMP.BaseElement {
     // from cache' issue.  See https://github.com/ampproject/amphtml/issues/5614
     /** @private {?ArrayBuffer} */
     this.creativeBody_ = null;
-    /** @private {?string} */
+    /**
+     * Note(keithwrightbos) - ensure the default here is null so that ios
+     * uses safeframe when response header is not specified.
+     * @private {?string}
+     */
```
Comment (lannka) [inline extensions/amp-a4a/0.1/amp-a4a.js]: what do you think if we initialize as:
```
this.experimentalNonAmpCreativeRenderMethod_ = platformFor(this.win).isIos() ? 'safeframe' : null;
```

Comment (keithwrightbos) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Done.  I was saving the isIos regexp execution if we got back the response header but I suppose we should assume going forward that would be a rarity.  I also had to duplicate this in unlayoutcallback to ensure we reset properly.  Let me know if you still think this is worth doing.

**Comment by keithwrightbos** [comment]
lannka@ responded to your comments.  If they look good, will you go ahead and merge?  Thanks

File: extensions/amp-a4a/0.1/test/test-amp-a4a.js [hunk 3]
```diff
@@ -182,6 +183,28 @@ describe('amp-a4a', () => {
       });
     });
 
+    it.only('for ios defaults to SafeFrame rendering', () => {
```
Comment (lannka) [inline extensions/amp-a4a/0.1/test/test-amp-a4a.js]: remove `.only`

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 4]
```diff
@@ -555,7 +561,8 @@ export class AmpA4A extends AMP.BaseElement {
       this.adPromise_ = null;
       this.adUrl_ = null;
       this.creativeBody_ = null;
-      this.experimentalNonAmpCreativeRenderMethod_ = null;
+      this.experimentalNonAmpCreativeRenderMethod_ =
+        platformFor(this.win).isIos() ? 'safeframe' : null;
```
Comment (lannka) [inline extensions/amp-a4a/0.1/amp-a4a.js]: nit: indent 2 more spaces

File: extensions/amp-a4a/0.1/amp-a4a.js (hunk 2 above)
Comment (lannka) [inline extensions/amp-a4a/0.1/amp-a4a.js]: that's fine. was thinking about avoiding the special "Note" in the JS doc.

**Comment by keithwrightbos** [comment]
@lannka sorry about that!  should be good now

**Pull request merged**